cluster/extensions/src/main/scala/com/lightbend/lagom/internal/cluster/projections/ProjectionRegistry.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package com.lightbend.lagom.internal.cluster.projections
+
+import akka.actor.ActorRef
+import akka.actor.ActorSystem
+import akka.actor.Props
+import akka.annotation.ApiMayChange
+import akka.pattern.ask
+import akka.cluster.sharding.ClusterShardingSettings
+import akka.util.Timeout
+import com.lightbend.lagom.internal.cluster.projections.ProjectionRegistry._
+import com.lightbend.lagom.internal.cluster.ClusterDistribution
+import com.lightbend.lagom.internal.cluster.ClusterDistributionSettings
+import com.lightbend.lagom.internal.cluster.projections.ProjectionRegistryActor.DesiredState
+import com.lightbend.lagom.internal.cluster.projections.ProjectionRegistryActor.GetDesiredState
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.concurrent.duration._
+
+@ApiMayChange
+object ProjectionRegistry {
+
+  sealed trait WorkerStatus
+  case object Stopped extends WorkerStatus
+  case object Started extends WorkerStatus
+
+  sealed trait StateRequest
+  case object Stop extends StateRequest
+  case object Start extends StateRequest
+
+  @ApiMayChange
+  case class ProjectionWorker(name: String, status: WorkerStatus)
+
+  @ApiMayChange
+  case class Projection(name: String, workers: Seq[ProjectionWorker])
+
+}
+
+@ApiMayChange
+private[lagom] class ProjectionRegistry(system: ActorSystem) {
+
+  // A ProjectionRegistry is responsible for this node's ProjectionRegistryActor instance.
+  // TODO: decide what to do if/when the ProjectionRegistryActor dies (note the loss of references to projections).
+  private val projectionRegistryRef: ActorRef = system.actorOf(ProjectionRegistryActor.props, "projection-registry")
```

Comment:
@renatocaval I'm not 100% what to do about this one. When the actor dies, some users may consider the node untrusted and may want to self-destruct (since projection workers are no longer inspectable).

The issue is: when `ProjectionRegistryActor` dies, the workers no longer have an `ActorRef` they can report to.

Reply:
We may want to configure it to resume in case of failure. In which case, we don't lose the actors current state. I don't want to hold the PR on this. Let's merge and we review it later.